src/NuGet.Core/NuGet.Resolver/PackageResolver.cs

```diff
@@ -42,6 +43,30 @@ public IEnumerable<PackageIdentity> Resolve(PackageResolverContext context, Canc
                 }
             }
 
+            // validate existing package.config for any invalid dependency
+            var invalidsPackagesLst = new List<string>();
+            var installedPackages = context.PackagesConfig.Select(p => p.PackageIdentity);
+            foreach (var package in installedPackages)
+            {
+                var existingPackage = context.AvailablePackages.FirstOrDefault(p => StringComparer.OrdinalIgnoreCase.Equals(p.Id, package.Id) && p.Version.Equals(package.Version));
+                if(existingPackage != null)
+                {
+                    // check if each dependency can be satisfied with existing packages
+                    var brokenDependencies = GetBrokenDependencies(existingPackage, installedPackages);
+                    if(brokenDependencies !=  null && brokenDependencies.Any())
+                    {
+                        invalidsPackagesLst.AddRange(brokenDependencies.Select(dependency => FormatDependencyConstraint(existingPackage, dependency)));
+                    }
+                }
+            }
+            // log warning message for all the invalid package dependencies
+            if(invalidsPackagesLst.Count > 0)
```

Comment:
It actually wont because the type is a list. 
Nits: format the document, add newlines after declarations and before if/for and don't use abbreviations like lst. Invalid packages is a good name, you hardly ever need to mention the type in the name of the variable